lib/methods/veres-one/client.js

```diff
@@ -0,0 +1,318 @@
+    const queryServiceUrl = url.parse(ledgerAgent.service.ledgerQueryService);
+    queryServiceUrl.search = `id=${did}`;
+    const didUrl = url.format(queryServiceUrl);
+
+    // const https = require('https');
+    // let agent;
+    // if(options.mode === 'dev') {
+    //   agent = new https.Agent({rejectUnauthorized: false});
+    // }
+    // _verbose(options, 'Retrieving remote DID Document...', {hostname});
+    // _debug(options, 'remote request', {url: didUrl});
+
+    const response = await r2({
+      url: didUrl,
+      method: 'POST',
+      // agent,
+      headers: {
+        'accept': 'application/ld+json, application/json'
+      }
+    }).response;
+
+    if(response.status !== 200) {
+      throw new Error('Failed to get DID Document; status=' + response.status);
+    }
+
+    const body = await response.json();
+
+    const result = {
+      found: true,
+      retry: false,
+      // timeMs,
+      // reqMs,
+      // start: reqOptions.start,
+      // retries: reqOptions.retries + 1,
+      type: 'LedgerDidDocument',
+      did,
+      // hostname,
+      doc: body.object,
+      meta: body.meta
+    };
+
+    return result;
+  }
+
+  /**
+   * Sends an operation to a Veres One ledger node.
+   *
+   * @returns {object} response
+   */
+  async send(options = {}) {
+    const ledgerAgent = await this.getAgent(options);
+    const {operation} = options;
+
+    const headers = options.headers ||
+      {
+        'content-type': 'application/ld+json, application/json',
+        'accept': 'application/json'
+      };
+
+    return r2({
+      url: ledgerAgent.service.ledgerOperationService,
+      method: 'POST',
+      // agent,
+      headers,
+      json: operation
+    }).response;
+  }
+
+  /**
+   * Resolves with a list of ledger agent urls.
+   *
+   * @param [options={}] {object}
+   * @param [options.mode]
+   * @param [options.hostname]
+   *
+   * @returns {Promise<Array<object>>}
+   */
+  async getAgents(options = {}) {
+    // const https = require('https');
+    // let agent;
+    // if(options.mode === 'dev') {
+    //   agent = new https.Agent({rejectUnauthorized: false});
+    // }
+    const hostname = this.singleHostname(options);
+    const ledgerAgentsUrl = `https://${hostname}/ledger-agents`;
+
+    const headers = {
+      'accept': 'application/ld+json, application/json'
+    };
+
+    const ledgerAgentRes = await r2({
+      url: ledgerAgentsUrl,
+      method: 'GET',
+      // agent,
+      headers
+    }).json;
+
+    return ledgerAgentRes.ledgerAgent;
+  }
+
+  /**
+   * Resolves with a single ledger agent for given options (first one).
+   *
+   * @param options {object} See docstring for `getAgents()` above.
+   *
+   * @returns {Promise<object>}
+   */
+  async getAgent(options) {
+    const agents = await this.getAgents(options);
+    return agents[0];
+  }
+
+  static signRequestHeaders({path, headers, signHeaders, keyId, key, method}) {
```

Comment:
I think we should be able to use this: https://github.com/digitalbazaar/http-signature-header/blob/master/lib/index.js

Here's a helper from another module that shows operation: https://github.com/digitalbazaar/bedrock-passport/blob/master/test/mocha/helpers.js#L19

`http-signature` is basically not being maintained and has no support for Ed25519 sigs (for instance).

Reply:
Thanks! I'm in the process of pulling out this code into its own web-ledger-client repo, and I'll use the suggested `http-signature-header` library there.

Reply:
@dmitrizagidulin does the change your referencing happen after the release we've discussed?